File: vlife-admin-react/src/mvc/model/FormField.ts
import { Schema } from "@formily/react";
import apiClient from "@src/mvc/apiClient";
import { PageVo, DbEntity, Result, VoBean } from "@src/mvc/base";
import { FormEventDto, FormEventVo } from "./FormEvent";
import { FormReactionVo } from "./FormReaction";

// 列表字段
export interface FormField extends DbEntity {
  formId: string; // 所属表单
  pathName: string; // 全路径
  fieldName: string; // 字段名
  description: string; // 描述
  readOnly: boolean; // 只读
  sort: number; // 顺序
  type: string; // 元素类型
  title: string; // 标题
  required: boolean; // 必填
  initialValues: string; // 默认值
  component: string; // 采用组件
  dictCode: string; // 字典的coden
  dataIndex: string; // formliy里的字段名称，默认一致
  x_decorator: string; // 包裹组件
  x_hidden: boolean; // 隐藏
  disabled: boolean; // 禁用
  fieldType: string; // 字段类型
  x_decorator_props$gridSpan: number; //包裹元素所占grid列数
  x_component_props$placeholder: string; //input填写说明
  x_decorator_props$layout: string; //标签位置
  x_decorator_props$labelAlign: string; //标签对齐方式
  x_validator: string; //校验方式
  vlife_pattern: string; //正则校验
  vlife_message: string; //校验不通过提醒
}

/**
 * 页面组件属性，期待全部干掉
 * 非数据库属性
 */
export interface CustomField {
  labelProps: any; //包装名称层属性
  props: {
    //自定义组件需要的属性
    params?: string[]; //loadData需要监听的属性，名称要修改
    loadData?: (obj?: any) => Promise<Result<any[]>>; //异步取数据promise方法把数据存入data
    // any各组件独有的属性
    datas?: any[]; //元素单个加载时候给的数据
    // syscParams?: string[] //异步的参数名称集合
    // syncDatas?: (obj: any) => Promise<Result<any>> //组件提取时给的数据
  }; //
}
/**
 * 字段属性
 */
export interface FormFieldVo extends VoBean, Schema, CustomField {
  [key: string]: any; // 实现可以赋值任意字段
  formId: string; // 所属表单
  pathName: string; // 全路径
  fieldName: string; // 字段名
  description: string; // 描述
  readOnly: boolean; // 只读
  sort: number; // 顺序
  type: string; // 元素类型
  title: string; // 标题
  required: boolean; // 必填
  initialValues: string; // 默认值
  x_component: string; // 采用组件
  dictCode: string; // 字典的coden
  dataIndex: string; // formliy里的字段名称，默认一致
  x_decorator: string; // 包裹组件
  x_hidden: boolean; // 隐藏
  disabled: boolean; // 禁用
  fieldType: string; // 字段类型
  x_decorator_props$gridSpan: number; //字段所占列数
  x_component_props$placeholder: string; //input填写说明
  x_decorator_props$layout: string; //标签位置
  x_decorator_props$labelAlign: string; //标签对齐方式
  apiKey: string; // 自定义展示组件需要调用的api关键字(名称)，去loadData.ts里换取实际api接口对象
  x_validator: string; //校验方式
  vlife_pattern: string; //正则校验
  vlife_message: string; //校验不通过提醒
  events: FormEventVo[]; //当前field触发的事件
  reactions: FormReactionVo[];
}

/**
 * 保存列表字段;
 * @param dto 列表字段;
 * @return 列表字段;
 */
export const save = (dto: FormField): Promise<Result<FormField>> => {
  return apiClient.post(`/formField/save`, { params: dto });
};
/**
 * 明细查询列表字段;
 * @param id null;
 * @return 列表字段;
 */
export const detail = (id: string): Promise<Result<FormField>> => {
  return apiClient.get(`/formField/detail/${id}`);
};
/**
 * 逻辑删除;
 * @param id null;
 * @return 已删除数量;
 */
export const remove = (id: string): Promise<Result<number>> => {
  return apiClient.delete(`/formField/remove/${id}`);
};

File: vlife-admin/src/main/java/cn/wwwlike/form/entity/FormField.java
package cn.wwwlike.form.entity;

import cn.wwwlike.vlife.annotation.VClazz;
import cn.wwwlike.vlife.bean.DbEntity;
import lombok.Data;

import javax.persistence.Entity;
import javax.persistence.Table;

/**
 * 模型字段信息
 * 也包含UI场景的配置信息
 */
@Entity
@Data
@Table(name = "form_field")
@VClazz(orders = "sort_asc",module = "conf",remove = {FormEvent.class,FormReaction.class})
public class FormField extends DbEntity {
    /**
     * 标题
     * db标题
     */
    public String title;
    /**
     * 标题
     * java标题
     */
    public String javaTitle;

    /**
     * 是否
     */
    public Boolean labelField;

    /**
     * 字段名
     */
    public String fieldName;

    /**
     * 前端数据类型
     * basic,array,object
     */
    public String dataType;
    /**
     * 字段类型
     */
    public String fieldType;
//    /**
//     * 元素类型
//     */
//    public String type;

    /**
     * 所属表单
     */
    public String formId;

    /**
     *  唯一不重复
     * validate是内置远程校验规则，触发form内部远程数据校验
     */
    public boolean  validate_unique;

    /**
     * 新增时隐藏
     */
    public boolean  create_hide;

    /**
     * 修改时只读
     * 不可修改
     */
    public boolean  modify_read;

    /**
     * 加入分割线
     */
    public boolean divider;
    /**
     * 分割线标签名称
     */
    public String dividerLabel;

    /**
     * 所在分组
     * 不存id原因，设计器阶段还没有产生id
     */
    public String formGroupCode;

    /**
     * 所在页签编码
     */
    public String formTabCode;

    /**
     * 标签隐藏
     */
    public boolean hideLabel;

    /**
     * 产生来源所在实体名称
     */
    public String entityType;

    /**
     * 字段来源实体对应字段名
     */
    public String entityFieldName;

    //组件类型
    public String componentType;

    /**
     * 全路径
     */
    public String pathName;

    /**
     * 字典的coden
     */
    public String dictCode;


    //-----------formily 场景字段配置的信息

    /**
     * formliy里的字段名称，默认一致
     */
    public String dataIndex;

    /**
     * 默认值
     */
    public String initialValues;
    /**
     * 必填
     */
    public boolean required;
    /**
     * 隐藏
     */
    public boolean x_hidden;
    /**
     * 描述
     */
    public String description;
    /**
     * 采用组件
     */
    public String x_component;
    /**
     * 只读
     *
     */
    public Boolean x_read_pretty;
    /**
     * 禁用
     */
    public Boolean disabled;
    /**
     * 只读
     */
    public Boolean readOnly;
    /**
     * 顺序
     */
    public Integer sort;
    //-------------外观
    /**
     * 包裹组件
     */
    public String x_decorator;

    /**
     * vlife开头 需要特殊处理的字段
     */
    public String vlife_pattern;

    public String vlife_message;

    /**
     * 对应 x-decorator-props.gridSpan属性
     */
    public Integer x_decorator_props$gridSpan;

    /**
     * label位置 水平 vertical
     */
    public String x_decorator_props$layout;

    /**
     * label位置 对齐 left
     */
    public String x_decorator_props$labelAlign;


    /**
     * 描述信息 对应 x-component-props.placeholder;
     */
    public String x_component_props$placeholder;
    /**
     * 加载组件数据的地址关键字
     */
    public String apiKey;
    /**
     * 字段校验方式
     */
    public String x_validator;
    /**
     * 最小长度
     */
    public Integer minLength;
    /**
     * 最小长度
     */
    public Integer maxLength;
    /**
     * 最小值
     */
    public Double minimum;
    /**
     * 最大值
     */
    public  Double maximum;

    /**
     * 是否列表展示
     */
    public Boolean listShow;

    /**
     * 组件设置的JSON信息
     */
    public String componentSettingJson;

}
